math/dec_test.go

```diff
@@ -1,1028 +1,479 @@
-			"\"-33499189745056880149688856635597007162669032647290798121690100488888732861290.034376435130433535\"",
-			"\"-33499189745056880149688856635597007162669032647290798121690100488888732861290.034376435130433535\"\n",
+		"Infinity": {
+			src:    "Infinity",
+			expErr: ErrInfiniteString,
 		},
-		{
-			math.LegacyNewDecFromBigIntWithPrec(maxInt, 18),
-			"3636373439353934383732353238343430303734383434343238333137373938353033353831333334353136333233363435333939303630383435303530323434343434333636343330363435303137313838323137353635323136373637",
-			"\"66749594872528440074844428317798503581334516323645399060845050244444366430645.017188217565216767\"",
-			"\"66749594872528440074844428317798503581334516323645399060845050244444366430645.017188217565216767\"\n",
+		"Inf": {
+			src:    "Inf",
+			expErr: ErrInfiniteString,
 		},
 	}
-
-	for _, tc := range testCases {
-		bz, err := tc.input.Marshal()
-		s.Require().NoError(err)
-		s.Require().Equal(tc.rawBz, fmt.Sprintf("%X", bz))
-
-		var other math.LegacyDec
-		s.Require().NoError((&other).Unmarshal(bz))
-		s.Require().True(tc.input.Equal(other))
-
-		bz, err = json.Marshal(tc.input)
-		s.Require().NoError(err)
-		s.Require().Equal(tc.jsonStr, string(bz))
-		s.Require().NoError(json.Unmarshal(bz, &other))
-		s.Require().True(tc.input.Equal(other))
-
-		bz, err = yaml.Marshal(tc.input)
-		s.Require().NoError(err)
-		s.Require().Equal(tc.yamlStr, string(bz))
+	for name, spec := range specs {
+		t.Run(name, func(t *testing.T) {
+			got, gotErr := NewDecFromString(spec.src, spec.constraints...)
+			if spec.expErr != nil {
+				require.ErrorIs(t, gotErr, spec.expErr, got.String())
+				return
+			}
+			require.NoError(t, gotErr)
+			assert.Equal(t, spec.exp.String(), got.String())
+		})
 	}
 }
 
-// Showcase that different orders of operations causes different results.
-func (s *decimalTestSuite) TestOperationOrders() {
-	n1 := math.LegacyNewDec(10)
-	n2 := math.LegacyNewDec(1000000010)
-	s.Require().Equal(n1.Mul(n2).Quo(n2), math.LegacyNewDec(10))
-	s.Require().NotEqual(n1.Mul(n2).Quo(n2), n1.Quo(n2).Mul(n2))
-}
-
-func BenchmarkMarshalTo(b *testing.B) {
-	b.ReportAllocs()
-	bis := []struct {
-		in   math.LegacyDec
-		want []byte
+func TestNewDecFromInt64(t *testing.T) {
+	specs := map[string]struct {
+		src       int64
+		constants []SetupConstraint
+		exp       string
+		expErr    error
 	}{
-		{
-			math.LegacyNewDec(1e8), []byte{
-				0x31, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30,
-				0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30,
-				0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30,
-			},
+		"zero value": {
+			src: 0,
+			exp: "0",
+		},
+		"positive value": {
+			src: 123,
+			exp: "123",
+		},
+		"negative value": {
+			src: -123,
+			exp: "-123",
+		},
+		"max value": {
+			src: 9223372036854775807,
+			exp: "9223372036854775807",
+		},
+		"min value": {
+			src: -9223372036854775808,
```

Comment:
nit: stdlib constants are easier to read
```suggestion
			src: math.MinInt64,
```